### ISSUE TITLE
What is the difference between doop and baobab-react's decorators?

### ISSUE THREAD

**Issue by eyy** [mistadikay/doob#2]
Hi there. Cool project. I've seen that baobab-react has some very similar [decorators](https://github.com/Yomguithereal/baobab-react#decorators). What are the differences?

Thanks

**Comment by mistadikay** [comment]
Hi @eyy! You're right, the idea of `DataInit` and `DataWatcher` decorators is very similar to the `root` and `branch` of baobab-react, and the work of @Yomguithereal is a huge influence.

What is different here is that besides slightly different API we wanted to go a little bit further by introducing `DataFetcher` decorator and probably going more towards Flux concepts of data flow in general. Also we're thinking about not using Baobab's `cursors` explicitly at all and probably not bonding tight with Baobab as the only possible option for immutable state.

The bottom line: as we mentioned in README, this project is a work in progress, but we have some interesting ideas, and hopefully we'll introduce them soon.
